src/SCForm.native.js

```diff
@@ -79,14 +80,15 @@ class SCForm extends Component {
   }
 
   onSubmit() {
-    const formData = this.form.getValue();
-    if (formData) {
-      Alert.alert('Submit Form', 'Would you like to submit this form?', [
-        { text: 'Cancel' },
-        { text: 'Submit', onPress: () => this.props.saveForm(formData) },
-      ]);
+    if (validateFields(this.form)) {
```

Comment:
Instead of passing the form ref here, I think we should pass the form value, and `this.state.options`. Then we can create a new options object with any new errors and return it. Then back in the component we could do `this.setState({options: newOptions})`. This way we don't have to mutate the form ref or call `forceUpdate`.